catalog/harvest/catalog-harvest-webdav/src/main/java/org/codice/ddf/catalog/harvest/webdav/WebDavHarvester.java

```diff
@@ -0,0 +1,163 @@
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.harvest.webdav;
+
+import com.github.sardine.Sardine;
+import com.github.sardine.SardineFactory;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.catalog.harvest.HarvestedResource;
+import org.codice.ddf.catalog.harvest.Harvester;
+import org.codice.ddf.catalog.harvest.Listener;
+import org.codice.ddf.catalog.harvest.common.FileSystemPersistenceProvider;
+import org.codice.ddf.catalog.harvest.common.HarvestedFile;
+import org.codice.ddf.catalog.harvest.common.PollingHarvester;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Polls an HTTP WebDav address. */
+public class WebDavHarvester extends PollingHarvester {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WebDavHarvester.class);
+
+  private final Set<Listener> listeners = new HashSet<>();
+
+  private final Sardine sardine = SardineFactory.begin();
+
+  private final String persistentKey;
+
+  private final DavAlterationObserver observer;
+
+  private final FileSystemPersistenceProvider persistenceProvider;
+
+  private final HarvestedResourceListener webdavListener;
+
+  /**
+   * Creates a WebDav {@link Harvester} which will harvest products from the provided address.
+   *
+   * @param address http URL WebDav address
+   * @param initialListeners {@link Listener}s to register to this harvester
+   */
+  public WebDavHarvester(String address, Set<Listener> initialListeners) {
+    super(5L);
+    Validate.notEmpty(address, "address {location} cannot be empty");
+
+    persistenceProvider = new FileSystemPersistenceProvider("harvest/webdav");
+    webdavListener = new HarvestedResourceListener();
+
+    initialListeners.forEach(this::registerListener);
+    persistentKey = DigestUtils.sha1Hex(address);
+    observer = getCachedObserverOrCreate(persistentKey, address);
+
+    super.init();
+  }
+
+  private DavAlterationObserver getCachedObserverOrCreate(String key, String rootEntryLocation) {
+    if (persistenceProvider.loadAllKeys().contains(key)) {
+      LOGGER.trace(
+          "existing webdav observer for persistence key [{}] found, loading observer", key);
+      return (DavAlterationObserver) persistenceProvider.loadFromPersistence(key);
+    }
+
+    LOGGER.trace(
+        "no existing webdav observer for persistence key [{}], creating new observer", key);
+    return new DavAlterationObserver(new DavEntry(rootEntryLocation));
+  }
+
+  @Override
+  public void poll() {
+    observer.addListener(webdavListener);
```

Comment:
❓Not sure I understand why we have to add and then remove the `webdavListener` here. We may want to add a comment to explain why.